src/backend/joanie/core/utils/contract_definition.py

```diff
@@ -104,16 +105,27 @@ def generate_document_context(contract_definition, user, order=None):
         try:
             organization_address = order.organization.addresses.get(is_main=True)
         except Address.DoesNotExist:
-            organization_address = ORGANIZATION_FALLBACK_ADDRESS
+            organization_address = None
 
         # Course
         course_code = order.course.code
         course_name = order.product.safe_translation_getter(
             "title", language_code=contract_definition.language
         )
         course_dates = order.get_equivalent_course_run_dates()
-        course_start = course_dates["start"]
-        course_end = course_dates["end"]
+        course_start = (
+            course_dates["start"].isoformat() if course_dates["start"] else None
```

Comment:
I believe we have an helper somewhere, but I can't find it.
It was something like 
```
def format_date(_date):
    return _date.isoformat() if _date else None
```

Reply:
ok, I found it [src/backend/joanie/edx_imports/utils.py](https://github.com/openfun/joanie/blob/main/src/backend/joanie/edx_imports/utils.py#L48)

Reply:
Oh yes, I have forgotten to set back the `.isoformat()` on `courses_dates['start'] `and `courses_dates['end']` in the generation of the context for a contract definition. My bad. In my last [PR](https://github.com/openfun/joanie/pull/614) I had a method that would prepare the dates in `isoformat()` before returning the values of datetimes objects in the adequate format. Thus, this method was taken away while reviewing which explains the bug.

+1 for @kernicPanel suggestion.

Reply:
You should use the DjangoJsonEncoder with the related json field : 
https://docs.djangoproject.com/en/5.0/topics/serialization/#django.core.serializers.json.DjangoJSONEncoder
https://docs.djangoproject.com/en/5.0/ref/models/fields/#jsonfield

Reply:
Or create your own encoder/decoder if you want